src/api/middleware/files.js

```diff
@@ -0,0 +1,110 @@
+const HTTPStatus = require("http-status-codes");
+const fs = require("fs");
+const path = require("path");
+const util = require("util");
+const { MulterError } = require("multer");
+const multerConfig = require("../../config/multer");
+const { ErrorTypes } = require("./errorHandler");
+const cloudinary = require("cloudinary").v2;
+const env = require("../../config/env");
+const ValidationReasons = require("./validators/validationReasons");
+
+
+const save_folder = path.join(__dirname, "../../../public/");
+if (!fs.existsSync(save_folder)) fs.mkdirSync(save_folder);
+
+const parseError = (message) => message.toLowerCase().replace(" ", "-");
+
+const single = (field_name) => (req, res, next) => {
```

Comment:
This name is unclear, what does this function do?

Reply:
The name mirrors the single function of `multer` (the package that processes `multipart/form-data`) and wraps it with error checking, the function is responsible for reading a **single** file from the request with the corresponding `filed_name`